File: FootScout-Vue.WebClient/src/models/interfaces/ClubOffer.ts
import type { PlayerAdvertisement } from './PlayerAdvertisement';
import type { OfferStatus } from './OfferStatus';
import type { PlayerPosition } from './PlayerPosition';
import type { UserDTO } from '../dtos/UserDTO';

// Przedstawia ofertę klubu dla zawodnika, zawierając informacje o ogłoszeniu, statusie oferty, pozycji, 
// wynagrodzeniu i nadawcy oferty.

export interface ClubOffer {
    id: number;
    playerAdvertisementId: number;
    playerAdvertisement: PlayerAdvertisement;
    offerStatusId: number;
    offerStatus: OfferStatus;
    playerPositionId: number;
    playerPosition: PlayerPosition;
    clubName: string;
    league: string;
    region: string;
    salary: number;
    additionalInformation: string;
    creationDate: string;
    clubMemberId: string;
    clubMember: UserDTO;
}

File: FootScout-Vue.WebAPI/Entities/ClubOffer.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace FootScout_Vue.WebAPI.Entities
{
    // Model (encja) oferty klubowej
    public class ClubOffer
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int PlayerAdvertisementId { get; set; }

        [ForeignKey("PlayerAdvertisementId")]
        public virtual PlayerAdvertisement PlayerAdvertisement { get; set; }

        [Required]
        public int OfferStatusId { get; set; }

        [ForeignKey("OfferStatusId")]
        public virtual OfferStatus OfferStatus { get; set; }

        [Required]
        public int PlayerPositionId { get; set; }

        [ForeignKey("PlayerPositionId")]
        public virtual PlayerPosition PlayerPosition { get; set; }

        [Required]
        [MaxLength(30)]
        public string ClubName { get; set; }

        [Required]
        [MaxLength(30)]
        public string League { get; set; }

        [Required]
        [MaxLength(30)]
        public string Region { get; set; }

        [Required]
        public double Salary { get; set; }

        [MaxLength(200)]
        public string AdditionalInformation { get; set; }

        [Required]
        [DataType(DataType.DateTime)]
        public DateTime CreationDate { get; set; }

        [Required]
        public string ClubMemberId { get; set; }

        [ForeignKey("ClubMemberId")]
        public virtual User ClubMember { get; set; }
    }
}